week3/src/App.js

```diff
@@ -0,0 +1,87 @@
+import logo from "./logo.svg";
+import { useState } from "react";
+import "./App.css";
+import FightGame from "./Fight";
+import CookJjin from "./img/CookJjin.png";
+import CoolJjin from "./img/CoolJjin.png";
+import CryJjin from "./img/CryJjin.png";
+import FuckJjin from "./img/FuckJjin.png";
+import HyumJjin from "./img/HyumJjin.png";
+import MacJjin from "./img/MacJjin.png";
+import ShyJjin from "./img/ShyJjin.png";
+import TropyJjin from "./img/TropyJjin.png";
+
+const fighterImgs = [
+  {
+    img: CookJjin,
+    name: "👨🏻‍🍳요리하는 남자👨🏻‍🍳 찌니💛",
+  },
+  {
+    img: CoolJjin,
+    name: "🐂황소같은 남자🐂 찌니💛",
+  },
+  {
+    img: CryJjin,
+    name: "💧눈물 흘리는 남자💧 찌니💛",
+  },
+  {
+    img: HyumJjin,
+    name: "먼혐지니👁👁",
+  },
+  {
+    img: MacJjin,
+    name: "💻맥북 가진남자💻 찌니💛",
+  },
+  {
+    img: ShyJjin,
+    name: "👀부끄러운 남자👀 찌니💛",
+  },
+  {
+    img: TropyJjin,
+    name: "🏆상받는 남자🏆 찌니💛",
+  },
+  {
+    img: FuckJjin,
+    name: "🎸뻐킹 멋진남자🎸 찌니💛",
+  },
+];
+
+window.onload = function () {
+  fighterImgs.sort(() => Math.random() - 0.5);
+};
+function App() {
+  window.onload();
```

Comment:
맨 처음 로드될 때 랜덤으로 이미지 섞는 거 아이디어 개멋져요.. 
js에서 사용했던대로 그 아이디어를 이렇게 가져와서 사용한 거 되게 멋진 것 같아요! 사실 어떻게든 동작하면 되는 거라 이렇게 해도 상관은 없는데, 우리가 React를 사용하는 이상 이 부분을 좀 더 React스럽게 구현하면 좋을 것 같아요 (React에서 window.onload와 같은 방식을 사용하지 않게 하기 위해 제공해주는 것들이 있거든요! )
```suggestion
function App() {
  useEffect(()=>{
    fighterImgs.sort(() => Math.random() - 0.5);
  },[])
```
React에서 제공하는 `useEffect`를 이용하면 특정 값이 바뀔 때 뿐만 아니라 이 `function App()`이라는 컴포넌트가 처음 로드될 때 하고 싶은 일을 하게할 수 있어요! 
```javascript
// 컴포넌트가 처음 로드될 때 무언가를 시키고 싶은 경우 
useEffect(()=>{
  컴포넌트 화면이 처음 로드될 때 하고 싶은 동작들
},[])

// 컴포넌트 내부에 어떤 변수 값이 변화했을 때 무언가를 시키고 싶은 경우 
useEffect(()=>{
  변수1, 변수2 값이 바뀌었을 때 행해져야하는 동작들 
},[변화를추적하고싶은변수1, 변화를추적하고싶은변수2])
```
이런 꼴로 useEffect를 잘 이용하면 컴포넌트가 처음 로드될 때, 혹은 어떤 변수 값이 변화했을 때 하고 싶은 동작을 바로 useEffect에서 실행시켜줘요! 이런 방식을 이용하는 것이 window.onload()를 이용하는 것 보다 좀 더 리액트를 잘 활용하는 방식인 것 같습니다 : )

Reply:
추가적으로 이 부분에 대해 좀 더 자세히 알고 싶다면 React의 lifecycle에 따른 useEffect의 사용 방식에 대해 찾아보면 좋을 것 같아요 !

Reply:
처음에 sorting 시켜줄 때 useEffect를 사용하면 되는 군요! 따로 함수를 정의해서 불러올 생각만했는데 좀더 react스러운 방법이 있네요! 참고하겠습니다!!